src/cdn_definitions/_impl/cdn_data.py

```diff
@@ -0,0 +1,26 @@
+from . import load_data, PathAlias
+
+
+class CdnData(object):
```

Comment:
About this CdnData class.  I think there's a decision to be made about the future of this library before we can know whether it's right to introduce this class:

For this library I think we have two main approaches how we could expose the data:

1. We could just return exactly what's loaded - a raw dict.
2. We return some kind of model objects which have been created from the underlying data, which could have additional validations, default values etc.

Currently, with rhui_aliases and origin_aliases, this library has gone for approach (2) as it doesn't expose the raw dict.  Instead it returns these PathAlias objects which have 'src', 'dest' attributes and apply some additional checks on top of just the schema.

Even though it's already done that way, I'm not entirely confident that's the best way and there's still a possibility of changing it before moving forward.  But whatever we do, we should be consistent and pick one approach or the other, we really don't want clients using a raw dict for some bits of data and objects for others.

I believe you'd be working on moving some data from cdn-utils into this project soon, so have you thought about what that will look like?  To take an example, `RELEASE_VERSIONS_BY_TYPE`: if we stick with (2), then we shouldn't just add a method/property which returns a raw `release_versions_by_type` dict.  Instead we'd be designing some kind of `ReleaseVersionMapping` class or similar, maybe adding an enum for all the valid types of aus, eus, tus, e4s.  Client may write something like: `CdnData().release_versions.for_type(ReleaseType.EUS)`.

I can't put my finger on exactly why but I'm starting to think that approach (2) might have been a mistake here and we should really just return the dict.  i.e. we migrate to approach (1) before going ahead with adding more cdn-utils data here.

Anyway, back to this CdnData class in this PR.  The impact on this class would be:

- If you want to continue with (2) then this class is basically right, and in the future it'll have to have a lot of additional properties/methods added, and other related classes created.
- Or if you agree with me that we should change direction and go for (1) - then I think this class either shouldn't exist, or it should be a private implementation detail (i.e. it's not imported into cdn_definitions/\_\_init\_\_.py).  The top-level rhui_aliases, origin_aliases functions could still exist temporarily, marked as deprecated, but going forward all clients are suggested to just use load_data which returns a plain dict.

Reply:
>  I believe you'd be working on moving some data from cdn-utils into this project soon, so have you thought about what that will look like? To take an example, RELEASE_VERSIONS_BY_TYPE: if we stick with (2), then we shouldn't just add a method/property which returns a raw release_versions_by_type dict. Instead we'd be designing some kind of ReleaseVersionMapping class or similar, maybe adding an enum for all the valid types of aus, eus, tus, e4s. Client may write something like: CdnData().release_versions.for_type(ReleaseType.EUS).

Yes, I did start fiddling around with creating a schema, set of reference data, and ReleaseVersionMapping class last week. I realized Python objects could potentially make it more difficult to extend the data in cdn-definitions-private. Say, for example, we want to add a new, super-secret "RUS" stream and for whatever reason we are forbidden from releasing any information about the existence of this stream to the public. If the rel-eng wants to configure the additional RUS stream, how would they do so without also adding an RUS attribute in the ReleaseType class? (I suppose they could add a "HIDDEN" attribute or something, but that's besides the point.) Whereas, if the data returned was a dictionary, so long as we allowed for additional properties on the schema's definition for Release Types, we could add an additional release type in our cdn-definitions-private/data.yaml that isn't dependent on a hard-coded set of releases in cdn-definitions. If we wanted to restrict the Release Types, we could instead impose that check by adding a test in cdn-definitions-private. That was a bad example, but hopefully you caught my gist and extrapolate beyond the example.

> I can't put my finger on exactly why but I'm starting to think that approach (2) might have been a mistake here and we should really just return the dict. i.e. we migrate to approach (1) before going ahead with adding more cdn-utils data here.

I agree.  I am having trouble imagining a scenario in which the python objects would offer enough benefit over a dictionary to justify the increased complexity and the potential for additional restraints. Option (1) it is.